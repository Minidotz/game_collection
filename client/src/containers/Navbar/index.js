import React from 'react';
import { AppBar, Toolbar, IconButton, Typography, Button, Avatar } from '@material-ui/core';
import { ArrowBack, Menu as MenuIcon} from '@material-ui/icons';

const Navbar = ({ title, showBack, goBack, toggleDrawer }) => {
    return (
        <AppBar>
            <Toolbar>
                <IconButton color="inherit" aria-label="Menu" onClick={showBack ? goBack : toggleDrawer}>
                    {showBack ? (
                        <ArrowBack />
                    ) : (
                        <MenuIcon />
                    )}
                </IconButton>
                <Typography variant="h6" color="inherit" style={{ flexGrow: 1 }}>
                    {title}
                </Typography>
                <div style={{ display: 'flex', alignItems: 'center' }}>
                    <Button disableRipple>
                        <Avatar style={{ backgroundColor: '#b02d27' }}>SP</Avatar>
                        <Typography variant="subtitle2" color="inherit" style={{ paddingLeft: '0.5em', textTransform: "none" }}>Stratos</Typography>
                    </Button>
                </div>
            </Toolbar>
        </AppBar>
    )
}

export default Navbar;