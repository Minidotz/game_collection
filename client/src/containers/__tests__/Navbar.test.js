import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';
import { ArrowBack, Menu } from "@material-ui/icons";
import { IconButton } from '@material-ui/core';

const component = renderer.create(<Navbar title="Test App" showBack={true} />);
it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('shows back button if showBack is true', () => {
    const instance = component.root;
    expect(instance.findByType(ArrowBack)).toBeTruthy();
});

it('shows menu button if showBack is false', () => {
    let component = renderer.create(<Navbar showBack={false} />);
    const instance = component.root;
    expect(instance.findByType(Menu)).toBeTruthy();
});

it('shows correct title when title prop is passed', () => {
    const instance = component.root;
    expect(instance.findByProps({variant: "h6"}).props.children).toBe("Test App");
});

it('fires goBack when back button is clicked', () => {
    const myMock = jest.fn();
    let component = renderer.create(<Navbar showBack={true} goBack={myMock} />);
    const instance = component.root;
    instance.findByType(IconButton).props.onClick();
    expect(myMock).toHaveBeenCalled();
});