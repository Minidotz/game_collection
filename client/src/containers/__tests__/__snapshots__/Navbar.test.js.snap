// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      aria-label="Menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
    </button>
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      Test App
    </h6>
    <div
      className="UserNavbarItem-root-92"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="MuiAvatar-root UserNavbarItem-avatar-93 MuiAvatar-colorDefault"
          >
            SP
          </div>
          <h6
            className="MuiTypography-root UserNavbarItem-nameText-94 MuiTypography-subtitle2 MuiTypography-colorInherit"
          >
            Stratos
          </h6>
        </span>
      </button>
    </div>
  </div>
</header>
`;
